acceptance/tests/facts/ruby.rb

```diff
@@ -1,49 +1,59 @@
-test_name "The Ruby fact should resolve as expected in AIO" do
+test_name "C100305: The Ruby fact should resolve as expected in AIO" do
   tag 'risk:high'
 
 #
 # This test is intended to ensure that the the ruby fact resolves
 # as expected in AIO across supported platforms.
 #
-
   skip_test "Ruby fact test is confined to AIO" if @options[:type] != 'aio'
 
-  step "Ensure the Ruby fact resolves as expected" do
+  require 'json'
+  require 'facter/acceptance/base_fact_utils'
+  extend Facter::Acceptance::BaseFactUtils
 
-    case agent['platform']
-      when /windows/
-        ruby_platform = agent['ruby_arch'] == 'x64' ? 'x64-mingw32' : 'i386-mingw32'
-      when /osx/
-        ruby_platform = /x86_64-darwin[\d.]+/
-      when /aix/
-        ruby_platform = /powerpc-aix[\d.]+/
-      when /solaris/
-        if agent['platform'] =~ /sparc/
-          ruby_platform = /sparc-solaris[\d.]+/
-        else
-          ruby_platform = /i386-solaris[\d.]+/
-        end
-      when /cisco_ios_xr/
-        ruby_platform = /x86_64-linux/
-      when /huaweios/
-        ruby_platform = /powerpc-linux/
-      else
-        if agent['ruby_arch']
-          ruby_platform = agent['ruby_arch'] == 'x64' ? 'x86_64-linux' : /(i486|i686|s390x)-linux/
+  agents.each do |agent|
+    step "Ensure the Ruby fact resolves as expected" do
+      case agent['platform']
+        when /windows/
+          ruby_platform = agent['ruby_arch'] == 'x64' ? 'x64-mingw32' : 'i386-mingw32'
+        when /osx/
+          ruby_platform = /x86_64-darwin[\d.]+/
+        when /aix/
+          ruby_platform = /powerpc-aix[\d.]+/
+        when /solaris/
+          if agent['platform'] =~ /sparc/
+            ruby_platform = /sparc-solaris[\d.]+/
+          else
+            ruby_platform = /i386-solaris[\d.]+/
+          end
+        when /cisco_ios_xr/
+          ruby_platform = /x86_64-linux/
+        when /huaweios/
+          ruby_platform = /powerpc-linux/
         else
-          ruby_platform = agent['platform'] =~ /64/ ? 'x86_64-linux' : /(i486|i686|s390x)-linux/
-        end
-    end
+          if agent['ruby_arch']
+            ruby_platform = agent['ruby_arch'] == 'x64' ? /(x86_64|powerpc64le)-linux/ : /(i486|i686|s390x)-linux/
+          else
+            ruby_platform = agent['platform'] =~ /64/ ? /(x86_64|powerpc64le)-linux/ : /(i486|i686|s390x)-linux/
+          end
+      end
 
-    ruby_version  = /2\.\d+\.\d+/
-    expected_ruby = {
-        'ruby.platform' => ruby_platform,
-        'ruby.sitedir'  => /\/site_ruby/,
-        'ruby.version'  => ruby_version
-    }
+      ruby_version   = /2\.\d+\.\d+/
+      expected_facts = {
+          'ruby.platform' => ruby_platform,
+          'ruby.sitedir'  => /\/site_ruby/,
+          'ruby.version'  => ruby_version
+      }
 
-    expected_ruby.each do |fact, value|
-      assert_match(value, fact_on(agent, fact))
+      step("verify that ruby structured fact contains facts") do
```

Comment:
Did this step get changed so we only talk to the agent once? Or was there some additional reason?